### PR TITLE
Correctly identify if we are in a static property.

### LINKED ISSUE #8010: Generate Field/Method not respecting static modifier on properties
Tested on Visual Studio 2015 Update 1. Appears to only affect static properties, not static methods.

### Steps to reproduce
```cs
using System;

public class Test
{
    public static int Property1
    {
        get { return _field; }
    }

    public static int Property2
    {
        get { return Method(); }
    }
}
```
Apply the Generate field and Generate method quick fixes.

### Expected result
```cs
using System;

public class Test
{
    private static int _field;

    public static int Property1
    {
        get { return _field; }
    }

    public static int Property2
    {
        get { return Method(); }
    }

    private static int Method()
    {
        throw new NotImplementedException();
    }
}
```

### Actual result
```cs
using System;

public class Test
{
    private int _field;

    public static int Property1
    {
        get { return _field; }
    }

    public static int Property2
    {
        get { return Method(); }
    }

    private int Method()
    {
        throw new NotImplementedException();
    }
}
```
Which does not compile, as `_field` and `Method` are not marked static.

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes #8010

### CODE REVIEW

**Comment by jmarolf** [comment]
retest vsi please

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs [hunk 1]
```diff
@@ -2782,6 +2782,24 @@ private static object DeserializeObject(JToken token, Type type)
...
+@"using System ; public class Test { public static int Property { get { return Method ( ) ; } } private static int Method ( ) { throw new NotImplementedException ( ) ; } } ");
+        }
+
+        [WorkItem(8010, "https://github.com/dotnet/roslyn/issues/8010")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)]
+        public async Task TestGenerateMethodFromStaticProperty2()
+        {
+            await TestAsync(
+@"using System ; public class OtherClass { } public class Test { public static OtherClass Property { get { if ( s_field == null ) s_field = [|InitializeProperty|] ( ) ; return s_field ; } } private static OtherClass s_field ; } ",
+@"using System ; public class OtherClass { } public class Test { public static OtherClass Property { get { if ( s_field == null ) s_field = InitializeProperty ( ) ; return s_field ; } } private static OtherClass InitializeProperty ( ) { throw new NotImplementedException ( ) ; } private static OtherClass s_field ; } ");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs]: So, to be clear.  The "inline" form of these tests was always meant for really really short examples.  When they go this long, teh intent was to write them out in long form.  :)

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs [hunk 2]
```diff
@@ -111,7 +111,12 @@ public static bool IsInStaticContext(this SyntaxNode node)
                     return memberDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword);
 
                 case SyntaxKind.PropertyDeclaration:
-                    return node.IsFoundUnder((PropertyDeclarationSyntax p) => p.Initializer);
+                    if (!memberDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword))
+                    {
+                        return node.IsFoundUnder((PropertyDeclarationSyntax p) => p.Initializer);
+                    }
+
+                    return true;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs]:
```
return memberDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword) || 
    node.IsFoundUnder((PropertyDeclarationSyntax p) => p.Initializer)
```

**Comment by brettfo** [comment]
Does this repro for VB, too?

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs (hunk 1 above)
Comment (dpoeschl) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs]: Yeah, this is really hard to understand. I was going to suggest changing from suffixing the test method names with numbers to using words to describe the difference, but then my head started hurting trying to understand nesting.

**Comment by dpoeschl** [comment]
@brettfo It already works correctly in VB:
```VB
Public Class Test
    Public Shared ReadOnly Property Property1 As Integer
        Get
            Return _field
        End Get
    End Property

    Public ReadOnly Property Property2 As Integer
        Get
            Return Method()
        End Get
    End Property
End Class
```

**Comment by dpoeschl** [comment]
:+1:

**Comment by jmarolf** [comment]
@brettfo VB is much simpler in this context, and as a result does not have this problem

**Comment by jmarolf** [comment]
retest vsi please

**Comment by dpoeschl** [comment]
Test Signoff - Looks good, didn't find any related issues.

**Comment by CyrusNajmabadi** [comment]
:+1:

**Pull request merged**